File: portal/portal/App_Code/DAO/CategoriaDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data;
using MySql.Data.MySqlClient;
using System.IO;

namespace portal.App_Code.DAO
{
    public class CategoriaDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public Categoria carregar(long pCodigo)
        {
            Categoria obj = new Categoria();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Categoria where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt32(registros["codigo"]);
                    obj.NomeTipo = registros["nome_tipo"].ToString();
                    obj.Descritivo = registros["descritivo"].ToString();                    
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<Categoria> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<Categoria> lista = new List<Categoria>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Categoria where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    Categoria obj = new Categoria();
                    obj.Codigo = Convert.ToInt32(registros["codigo"]);
                    obj.NomeTipo = registros["nome_tipo"].ToString();
                    obj.Descritivo = registros["descritivo"].ToString();
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public static List<Categoria> carregarLista(Double pConteudo)
        {
            List<Categoria> lista = new List<Categoria>();
            try
            {
                String sc = Properties.Settings.Default.CN;
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select t1.* from Categoria t1,conteudo_categoria t2 where t1.codigo = t2.codigo_categoria and t2.codigo_conteudo=" + pConteudo.ToString();
                
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    Categoria obj = new Categoria();
                    obj.Codigo = Convert.ToInt32(registros["codigo"]);
                    obj.NomeTipo = registros["nome_tipo"].ToString();
                    obj.Descritivo = registros["descritivo"].ToString();
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public void persistir(Categoria obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
                if (obj.Codigo == 0)
                {
                    sql = "insert into Categoria(nome_tipo, descritivo) value('{0}','{1}')";
                    sql = String.Format(sql, obj.NomeTipo, obj.Descritivo);
                }
                else
                {
                    sql = "update Categoria set nome='{0}', email='{1}'  where codigo={2}";
                    sql = String.Format(sql, obj.NomeTipo, obj.Descritivo);
                }

                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }


        public void remover(Categoria obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "delete from Categoria where codigo=" + obj.Codigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

    }
}

File: portal/portal/App_Code/DAO/CategoriaDAO.php
<?php
namespace portal\App_Code\DAO {
    
//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------

use portal\App_Code\Categoria;     
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);

    try{
        require_once("/../Categoria.php");        
        require_once("/../../connection.php");        
    }catch(Exception $e){
        require_once (dirname(__file__, 2) . "\Categoria.php");
        require_once (dirname(__file__, 3) . "\connection.php");
    }
    
    class CategoriaDAO{	
//INSERIR----------------------------------------------------------------------------------------------------------------------------------------
    function inserirCategoria($Categoria){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into categoria (nome_tipo, descritivo) values (' " .
                 $Categoria->getNomeTipo() . " ',' " .
                 $Categoria->getDescritivo() . " ')";
                 mysqli_query($con,$sql);
				 echo "Inserção com sucesso!";
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
        }
    mysqli_close($con);
	}
//ALTERAR---------------------------------------------------------------------------------------------------------------------------------------
    function alterarCategoria($Categoria){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update categoria set " . 
                "nome_tipo = ' " . $Categoria->getNomeTipo() . " ' ,
                descritivo = ' " . $Categoria->getDescritivo() . " ' 
                where codigo = " . $Categoria->getCodigo() ;
                mysqli_query($con,$sql);
				echo "Dados alterados com sucesso!";
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                }
    mysqli_close($con);	
    }

//CARREGA OBJETO-------------------------------------------------------------------------------------------------------------------------------
    function carregarCategoria ($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from categoria where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
						$categoria = new Categoria();
                        $categoria->setCategoria(($carga["codigo"]),
												 ($carga["nome_tipo"]),
												 ($carga["descritivo"]));
                        mysqli_close($con);
                        return $categoria;
                    }else {
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;
            }
        mysqli_close($con);
    }

//CARREGA LISTA OBJETO 1----------------------------------------------------------------------------------------------------------------------

    function carregarListaCategorias($qtdOpcoes, $ordem){//qtdOpcoes = vetor | ordem = order by desc / asc
        
        $sql = "select * from categoria where ";
        
            for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++ ){
                $sql = $sql . $qtdOpcoes[$i];
				    if($i != $qtdOpcoes[$i] - 1){
                        $sql = $sql . "and ";  
                    } 
            }
            
            $sql = $sql . " order by " . $ordem;
        
            try{
                $con = BD_AbrirConexao();
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
							 $listaCat = [];
                       while($carga = mysqli_fetch_array($gargantula)){
                             $categoria = new portal\App_Code\Categoria();
							 $categoria->setCategoria(($carga["codigo"]),
													  ($carga["nome_tipo"]),
													  ($carga["descritivo"]));
                             $listaCat[] = $categoria;
                       }
                       mysqli_close($con);
                       return $listaCat;
                    }else {
                        echo "Registros não encontrados!";
                     }
            }catch(Exception $e){
                echo "Erro ao carregar dados: " . $e;
            }
        mysqli_close($con);
    }

//CARREGA LISTA OBJETO 2----------------------------------------------------------------------------------------------------------------------
    function carregarListaCategoriaCod($dado){
        
        $sql = "select t1.* from categoria t1, conteudo_categoria t2 where t1.codigo = t2.codigo_categoria and t2.codigo_conteudo = " . $dado;
         
        try{
            $con = BD_AbrirConexao();
            $gargantula = mysqli_query($con,$sql);
                if($carga = mysqli_fetch_array($gargantula)){
						  $listaCat = [];
                       while($carga = mysqli_fetch_array($gargantula)){
                             $categoria = new portal\App_Code\Categoria();
							 $categoria->setCategoria(($carga["codigo"]),
													  ($carga["descritivo"]),
													  ($carga["nome_tipo"]));
                             $listaCat[] = $categoria;                      
                    }
                    mysqli_close($con);
                    return $listaCat;                    
                }else {
                        echo "Registros não encontrados!";
                }
         }catch(Exception $e){
                echo "Erro ao carregar dados: " . $e;
         }
        mysqli_close($con);        
    }

//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarCategoria($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from categoria where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
				echo "Exclusão com sucesso";
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;
            }
        mysqli_close($con);
    }
  }
}
?>